src/java/test/org/apache/zookeeper/test/NettyLocalSuiteTest.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.zookeeper.test;
+
+import org.junit.runners.Suite;
+
+/**
+ * Run tests with: Netty Client against Netty server
+ */
+@Suite.SuiteClasses({
```

Comment:
> But IMHO a new system property/run time flag will add the need to run always twice the full suite at every build/release.

Maybe. We have a good degree of flakyness in our testing and wondering if using this type of change for our precommit  hook will increase performance and stability (especially if we can use something similar for server<->server).

> At this moment I think it is not the time to add an official "local transport" netty connection factory.

Agreed.

> I can't find how to enable Netty for server-to-server communications

It is currently not possible. All communications there are handled by old fashioned sockets.  I was incorrect in my other comment, although I still think my point is valid. Is there a way to have similar functionality with old fashioned java sockets? I don't think this change makes too much sense for testing unless we can take zk off the OS network stack entirely.